### PR TITLE
Fix reward net shrinking episode length bug

### PR DESCRIPTION
**Pull request by bkuen**

### CODE REVIEW

File: framework/sac_rlhf.py [hunk 1]
```diff
@@ -489,7 +489,7 @@ def get_action(self, x):
         env=envs,
     ).to(device)
     reward_optimizer = optim.Adam(
-        reward_net.parameters(), lr=args.teacher_learning_rate
+        reward_net.parameters(), lr=args.teacher_learning_rate, weight_decay=1e-4
```
Comment (bkuen) [inline framework/sac_rlhf.py:492]: This is the crutial part: `weight_decay=1e-4 ` adds a small L2 penalty on the model parameters. It directly penalizes large weights, encouraging the network to find simpler (smoother) solutions

File: framework/reward_net.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def gen_reward_net(hidden_dim, layers, env=None):
         reward_net.append(nn.Linear(hidden_dim, hidden_dim))
         reward_net.append(nn.LeakyReLU())
     reward_net.append(nn.Linear(hidden_dim, 1))
-    reward_net.append(nn.Tanh())
+    # reward_net.append(nn.ReLU())
```
Comment (bkuen) [inline framework/reward_net.py:22]: I think it's not a good idea to use an activation function in the last layer because this can cause the model to gravitate toward values near zero if it’s not strongly trained toward positives.

**Comment by Wizzardexe** [comment]
I generated some videos for that change. These look quite promising: 

https://github.com/user-attachments/assets/65592987-3e6f-4871-b366-b3391d9151ba

https://github.com/user-attachments/assets/86827558-098c-42c7-92fe-717d4552a466

**Comment by Wizzardexe** [comment]
Screenshot 2024-12-23 180549
This really looks great in Hopper-v4 combined with my pull request.

**Review by Wizzardexe — APPROVED** [review]

**Review by Wizzardexe — APPROVED** [review]
